### DISCUSSION TITLE
Control Thermostat via Temp Offset

### DISCUSSION THREAD

**Discussion by Caliban2017** [jmcollin78/versatile_thermostat#278] [Ideas]
Hello,

i am new to Versatile Thermostat and smart Valves. I bought some Tuya TS601 to begin with and installed them and they work fine so far. I first tried them with Better Thermostat but it is way to agressive in controlling the Valves. It always open all Valves to 100% for to long and overshot the Target Temp for 2-3°C or it goes completly to 0% and the Radiators are cooling out. I switched then to a Blueprint to match the Temperatures from a external Sensor, it works out fine on one single Thermostat but i have four in my Living Room and there i have the oppisite, it needs 3-4 Hours to reach a desired Temperature.

I now discovered Versatile Thermostat but i don't understand some Things here. What exactly is here possible? I set them to over_climate , i chosed a external Temp Sensor for the Room and a Outside Temp Sensor. But when i activate the Self-Regulation, it put 2°C on top of the Target Temperature to reach the Setpoint Temp. But the Problem here is, the Thermostat itself is also Set to this Temperature. So when i want 21°, Versatile Thermostat set all Thermostats to 23°C Target Temperature. This maybe works to reach the Temperature but in this Way i can not change the Temperatur on the Thermostats itself?!

So is it possible to control the regulation over the Offset-Temperatur, which this Devices Support? In this case it would be possible to set the Temperatur over the Device Itself and not only via Home Assistant.

Or do i understand something wrong here?

Have anyone experience with the Tuya TS601 in combination with Versatile Thermostat and can tell me the best settings for it?

Thank you and greetings,
Caliban

**Comment by jmcollin78** [comment]
Hello @Caliban2017,

You have a good understanding on how it works.

> This maybe works to reach the Temperature but in this Way i can not change the Temperatur on the Thermostats itself?!

You always can change the target temperature on the thermostat itself and VTherm will set its internal target temp (before regulation to this point). Then it will applied regulation and send the regulated temp to the TRV. So you should see the target temperature changing due to this regulation. It you change the target temp on the TRV, the VTherm will goes to the manual mode (not with presets).

So this is possible but not recommended: you preferably let VTherm regulate based on preset and don't touch the TRV itself. If your schedule and regulation works fine, you wont have to change the TRV itself. (but you still can do it for temporary boost or whatever).

This part have been difficult to implement because it is possible to have contradictory control command from human and VTherm. That is why it is not recommended (but possible).

> So is it possible to control the regulation over the Offset-Temperatur, which this Devices Support?

What is this offset temperature supported by the device ? How it works ?

**Comment by Caliban2017** [comment]
Hello again,

thank you for your Explanation so far, i think i understand it so far. So, the Offset Temperature on this Device can change the internal Temperatursensor by a given Offset. So when the Device say it is 21°C in the Room, you can give a Offset of -3 for example and it corrects the internal Temperature to 21-3=18°C. So the TVR think it is 18°C in the Room. The Offset is calculated by the External Temp Sensor. So when the Thermostat say it is 22° and the Room Sensor say it is 20°, the Offset is -2 and will be substracted from the TVR internal Temp, so it will begin to regulate the Temp. In this Way the Target Temperature on the Device itself can stay untouched, so you can still change it on the Device, because the Self-Regulation Correction is only happening on the internal Temperature of the Device. It seems that Better Thermostat is working exactly this way but the Regulation is way to strong there, there is no option to regulate it like in Versatile Thermostat with a Light/Medium/Strong Preset or even a manual Preset.

i hope i understand everything correct here, i am still new to smart Thermostats etc and try to learn and understand it.

You can find the Offset here:

![offset](https://github.com/jmcollin78/versatile_thermostat/assets/32013635/33ff1aea-8658-42ba-b21c-2ad9bb336b09)

**Reply by LennardPlay** [reply]
A temperature offset actually is really common with TRVs or room thermostats in general, so utilizing that option would be great for a lot of users. It would be a help for family members/guest wondering about the difference between the dashboard and TRV temperature. I guess it could also lead to some confusion, because you wouldn't be able to troubleshoot the VTherms regulation as easily in the history tab of HA.

VTherm already is calculating a temperature offset internally (before adding it to the target temperature) so maybe this could be a relatively easy to implement feature. (Says me, who already gives up when facing complex yaml 😅)

@jmcollin78 I would love to see this added, but I totally understand if that's not something you're interested in adding, or think that adds too much complexity. Thank you a lot for this wonderful integration 👍

**Reply by jmcollin78** [reply]
Hello @LennardPlay,

I don't remember where I write this, but VTherm is calculating itself an offset which in real time depending on the actual conditions. This is for me a great way to deal with offset because the offset is not static. When it is very cold outside and when it is hot the offset cannot be the same.
So with VTherm there is no need to configure an offset on the device but that makes no difference for VTherm. The internal real time calculated will adjust itself if you set an offset to your TRV. That is for VTherm just useless (but can be useful if you want to use the internal TRV temperature value).

> @jmcollin78 I would love to see this added, but I totally understand if that's not something you're interested in adding, or think that adds too much complexity.

So regarding this I will not add a static offset which is just totally useless for VTherm (this is not only a matter of complexity this time).

**Reply by LennardPlay** [reply]
I think I misunderstood what @Caliban2017  was talking about. I thought he was suggesting using the temperature_offset entity for dynamic regulation (wich I think might be a good idea).

I will open a feature request and describe my idea in more detail.

**Reply by LennardPlay** [reply]
I've opened up a discussion about my idea: https://github.com/jmcollin78/versatile_thermostat/discussions/342
Maybe that's interesting to you
